File: OpenBook/source/co_routine.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int is_Sub_String(char *string, char *subString){

}

int print_name(char *prefix){
    static int state = 0;
    char *string, *p;
    switch(state){
        case 0:
            string = prefix;
            state = 1;
            printf("Searching for prefix : %s\n", string);
            return 0;
        case 1:
            p = strstr(prefix, string);
            if(p){
                printf("%s\n", prefix);
            }
            return 0;
    }
}

int main(){
    print_name("Dear");
    print_name("Dear Mohith");
    print_name("Dr Mohith");
    print_name("Dears");

    return 0;
}

File: OpenBook/source/co_routine.py
def print_name(prefix):
	print("Searching prefix:{}".format(prefix))
	try :
		while True:
			name = (yield)
			if prefix in name:
				print(name)
	except GeneratorExit:
		print("Closing coroutine!!")

def main():
	handle = print_name("Dear") 		# Call the co Routine first
	handle.__next__()					# Move to next
	handle.send("Mahesh")				# Send a message to co routine
	handle.send("Dear Mahesh") 			# Send a message to co routine
	handle.send("Dear Anand")			# Send a message to co routine
	handle.send("Dr Anand")				# Send a message to co routine
	handle.close()						# Close the handle

if __name__ == '__main__' :
	main()
